tests/test_ensemble.py

```diff
@@ -0,0 +1,381 @@
+        self.assertAlmostEqual(ens.V.volume, 27.0)
+        self.assertDictEqual(ens.mu.todict(), {'A':0.1,'B':None})
+        self.assertDictEqual(ens.N.todict(), {'A':None,'B':2})
+        self.assertDictEqual(ens.constant, {'P':False,
+                                            'V':True,
+                                            'mu':{'A':True,'B':False},
+                                            'N':{'A':False,'B':True}})
+        self.assertAlmostEqual(ens.beta, 0.01)
+
+        #test creation with single type
+        ens = relentless.Ensemble(T=100, V=v_obj, mu={'A':0.1})
+        self.assertCountEqual(ens.types, ('A',))
+        self.assertCountEqual(ens.rdf.pairs, (('A','A'),))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 100)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V.volume, 27.0)
+        self.assertDictEqual(ens.mu.todict(), {'A':0.1})
+        self.assertDictEqual(ens.N.todict(), {'A':None})
+        self.assertDictEqual(ens.constant, {'P':False,
+                                            'V':True,
+                                            'mu':{'A':True},
+                                            'N':{'A':False}})
+        self.assertAlmostEqual(ens.beta, 0.01)
+
+        #test setting rdf
+        ens = relentless.Ensemble(T=100, V=v_obj, mu={'A':0.1}, N={'B':2})
+        r = [1,2,3]
+        g_ab = [2,9,5]
+        g_aa = [3,7,4]
+        g_bb = [1,9,3]
+        ens.rdf['A','B'] = relentless.RDF(r=r, g=g_ab)
+        ens.rdf['A','A'] = relentless.RDF(r=r, g=g_aa)
+        ens.rdf['B','B'] = relentless.RDF(r=r, g=g_bb)
+        np.testing.assert_allclose(ens.rdf['A','B'].table, np.array([[1,2],
+                                                                     [2,9],
+                                                                     [3,5]]))
+        np.testing.assert_allclose(ens.rdf['A','A'].table, np.array([[1,3],
+                                                                     [2,7],
+                                                                     [3,4]]))
+        np.testing.assert_allclose(ens.rdf['B','B'].table, np.array([[1,1],
+                                                                     [2,9],
+                                                                     [3,3]]))
+
+        #test invalid setting of neither P nor V
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(T=10, mu={'A':0.1,'B':0.2})
+
+        #test invalid setting of neither mu nor N
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(T=10, P=2.0)
+
+        #test invalid setting of both P and V
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(T=10, P=2.0, V=v_obj)
+
+        #test invalid setting of both mu and N
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(T=10, mu={'A':0.1,'B':0.2}, N={'A':1,'B':2})
```

Comment:
Can you add another test like this but where you do one of the pairs correctly and the other pair incorrectly?